pkg/resource/unstructured/unstructured.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unstructured
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// An ObjectTyper returns the GroupVersionKind that a runtime.Object considers
+// itself to have.
+type ObjectTyper struct{}
+
+// ObjectKinds returns one GroupVersionKind of the supplied object. It always
+// return false and nil.
```

Comment:
We have a few types that use a `runtime.ObjectTyper` (i.e. a `scheme.Scheme`) to determine the GVK of an object, because not all `runtime.Objects` have useful `GetObjectKind()` methods. This allows those types to accept an `ObjectTyper` that just returns `GetObjectKind` under the hood.

Reply:
Actually `scheme.Scheme` already handles this: https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/scheme.go#L251

Reply:
Oh nice - will remove.